spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java

```diff
@@ -121,6 +121,15 @@ public CloudSqlJdbcInfoProvider defaultMySqlJdbcInfoProvider(GcpCloudSqlProperti
 						+ defaultProvider.getJdbcDriverClass());
 			}
 
+			if (System.getenv("VCAP_SERVICES") != null) {
```

Comment:
did you consider using [Environment](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/env/Environment.html) instead? That should make it easier to test too! 
+1 for extracting string literal to constant here as well